src/components/Sidebar.tsx

```diff
@@ -0,0 +1,134 @@
+import { t } from "../i18n";
+import { useDevice } from "./App";
+import "./Sidebar.scss";
+import { SidebarLockButton } from "./SidebarLockButton";
+import { close } from "./icons";
+import React, { useContext, useLayoutEffect, useRef } from "react";
+import clsx from "clsx";
+import { Island } from "./Island";
+import { atom, useAtom } from "jotai";
+import { jotaiScope } from "../jotai";
+import { withUpstreamOverride } from "./hoc/withUpstreamOverride";
+
+/** using a counter instead of boolean to handle race conditions where
+ * the host app may render (mount/unmount) multiple different sidebar */
+export const hostSidebarCountersAtom = atom({ rendered: 0, docked: 0 });
+
+const _SidebarHeader: React.FC<{
```

Comment:
actually, even worse, as we'd need to export the `_SidebarHeader` to import it into the `Sidebar.tsx` file, where we'd then wrap it into `withUpstreamOverride()`, where we'd then namespace it the `Sidebar` component.

We'd gain nothing much except cyclic dependency (something I want to disallow later), polluting import space & make it more complex.

Reply:
> We want to make it clear the _SidebarHeader is not to be used anywhere, even inside our codebase. So underscoring it makes sense here. We can also add @private, but that won't help much when importing.

I think this shouldn't be a concern unless you are worried for some specific reason and also eventually we want to have a `ui-kit` of our own where we would be moving the component. Also as per usage in our codebase this is not like those general ui components (eg button, dropdown) which could be used anywhere and is very specific to sidebar.

I think we could create `sidebar` folder and inside it have `Sidebar.tsx` and `Header.tsx` files.

Reply:
> I think this shouldn't be a concern unless you are worried for some specific reason and also eventually we want to have a `ui-kit` of our own where we would be moving the component.

I have a specific reason: I don't want us to mistakenly use it in our codebase (happens a lot in E+ codebase with components that share similar or identical names to node_modules components) — we'd have to set up eslint rules for it. And it's also gonna pollute intellisense (no rules can be made against that).

Is there a specific reason you want to avoid underscording?

> I think we could create `sidebar` folder and inside it have `Sidebar.tsx` and `Header.tsx` files.

What I said above still applies:

> We'd gain nothing much except a cyclic dependency (something I want to disallow later), polluting import space & make it more complex.